test_libs/pyspec/eth2spec/utils/ssz/test_ssz_partials.py

```diff
@@ -0,0 +1,112 @@
+from eth2spec.utils.ssz.ssz_impl import serialize, hash_tree_root
+from eth2spec.utils.ssz.ssz_typing import (
+    boolean, Bytes, byte, uint64, Container, Vector, List
+)
+
+from eth2spec.utils.ssz.ssz_partials import (
+    ssz_full, ssz_partial, extract_value_at_path, merge
+)
+
+
+class Person(Container):
+    is_male: boolean
+    age: uint64
+    name: Bytes[32]
+
+
+class City(Container):
+    coords: Vector[uint64, 2]
+    people: List[Person, 20]
+
+
+people = List[Person, 20](
+    Person(is_male=True, age=uint64(45), name=Bytes[32](b"Alex")),
+    Person(is_male=True, age=uint64(47), name=Bytes[32](b"Bob")),
+    Person(is_male=True, age=uint64(49), name=Bytes[32](b"Carl")),
+    Person(is_male=True, age=uint64(51), name=Bytes[32](b"Danny")),
+    Person(is_male=True, age=uint64(53), name=Bytes[32](b"Evan")),
+    Person(is_male=False, age=uint64(55), name=Bytes[32](b"Fae")),
+    Person(is_male=False, age=uint64(57), name=Bytes[32](b"Ginny")),
+    Person(is_male=False, age=uint64(59), name=Bytes[32](b"Heather")),
+    Person(is_male=False, age=uint64(61), name=Bytes[32](b"Ingrid")),
+    Person(is_male=False, age=uint64(63), name=Bytes[32](b"Kane")),
+)
+
+city = City(coords=Vector[uint64, 2](uint64(45), uint64(90)), people=people)
```

Comment:
Can we do `city = City(coords=[45, 90], people=people)` now due to auto-type-coercion?